src/components/carousel/index.tsx

```diff
@@ -437,17 +441,17 @@ class Carousel extends Component<CarouselProps, CarouselState> {
       : undefined;
     const snapToOffsets = this.getSnapToOffsets();
     const marginBottom = Math.max(0, this.getContainerPaddingVertical() - 16);
-
     return (
       <View animated={animated} style={[{marginBottom}, containerStyle]} onLayout={this.onContainerLayout}>
         <ScrollView
           {...others}
           ref={this.carousel}
           contentContainerStyle={scrollContainerStyle}
-          horizontal
+          horizontal={this.props.horizontal}
           showsHorizontalScrollIndicator={false}
-          pagingEnabled={this.shouldEnablePagination()}
-          snapToOffsets={snapToOffsets}
+          showsVerticalScrollIndicator={false}
+          pagingEnabled={this.props.horizontal ? this.shouldEnablePagination() : true}
```

Comment:
Consider moving this condition inside the `shouldEnablePagination` method. 
This will also fix the next line (: